### DISCUSSION TITLE
order code tracking - is this possible

### DISCUSSION THREAD

**Discussion by Mistaken4now** [nextgenhealthcare/connect#5229] [Q&A]
I am hoping you can point me in the right direction

I have a system that "normalizes" order descriptions.  the facility we send the orders back wants their description back not our normalized code. 

the go to answer is a cross walk to change things back.  (lets say this method has some draw back in this case)

SO . . .  I was hoping to do this more I could put the order number and description in a mirth variable and what for the report to come back out, and match up on order number and write the description back in.

The catch being if Mirth stores this in memory and I have to restart mirth I would loose my look up array.

1) can I send something to a giant 2d global array?
2) can I also send it to a static file
3) can I check to make sure said 2d global array is initialized and if not call a function to read in the static file?
4) can I querry that global array (assuming if I can create it I can query or update it)
5) saying ALL of the above can be done is there a gotcha I am missing

Thank you so much.

**Comment by pacmano1** [comment]
There are several options - as you mentioend though restarting mirth would remove something you created on the fly in a global map or global channel map.  You can store objects in mirth maps like $g and $gc.

In your use case, why not just create that lookup structure in a database table?   Given the correct design on your side that may be an "extra" call to crosswalk back to the orginal recieved item given the crosswalk could be returned in the source (e.g. a db reader).  

If these are inbound orders - I have seen some cool solutions that go back into the orginal receiving channel to get the exact message that was received to handle the crosswalk.   

Anoter option is a guava cache, which ships with mirth.

**Comment by siddharthsharma9258** [comment]
I would suggest doing File I/O or use a database table for this kind  of requirement.
